esmtools/stats.py

```diff
@@ -474,18 +475,32 @@ def xr_autocorr(da, lag=1, dim='time', r_only=False):
     if r_only:
         return r
     else:
-        return r, p 
+        return r, p
+
+def xr_corr(ds, lag=1, dim='year'):
+    """
+    Calculated lagged correlation of a xr.Dataset.
+
+    This is a faster implementation that scipy pearsonr. Uses xskillscore's pearson_r.
+    """
+    from xskillscore import pearson_r
```

Comment:
Move this import statement to the header where all the other import statements are. This helps avoid importing multiple times and makes sure that the import starts at the beginning of the file.